File: day2/src/part1.py

filename = '..\data\data.txt'
with open(filename) as fn:
    location = [0, 0]
    ln = fn.readline()
    while ln:
        lnsplit = ln.split()
        if lnsplit[0] == 'forward':
            location[0] += int(lnsplit[1])
        else:
            if lnsplit[0] == 'down':
                location[1] += int(lnsplit[1])
            else:
                location[1] -= int(lnsplit[1])
        ln = fn.readline()
    print(location[0]*location[1])

File: day2/src/part1.cpp
#include <fstream>
#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

vector<string> data;
int count;

int readFile(string filename, vector<string>& vec)
{
    string line;
    int i, num;
    stringstream stream;
    ifstream reader(filename);

    for(i=0;!reader.eof();i++)
    {
        getline(reader,line);
        vec.push_back(line);
    }
}

int getSubstring(string lineInput, int placement)
{
    if (placement ==1){
        // return magnitude integer
        return 0;
    } else {
        // return direction integer
        return 0;
    }
}

int processString(string lineInput)
{
    // or use array here
    int dirmag[2];
    dirmag[0] = direction; 
    dirmag[1] = magnitude;
    return result {getSubstring(lineInput,0),getSubstring(lineInput,1)};
}


int main(int argc, char** argv){

    readFile("../data/sample.txt", data);
    int i;
    for(i=0; i < data.size(); i++)
    {
        cout << data[i] << endl;
    }

}